### PR TITLE
Fix watcher errors

### LINKED ISSUE #16: Watch not working in vue files if this plugin is enabled
If I have Single Components and multiple files components in the same project, like:
```
Example.Vue
Example2/
    Example2.js
    Example2.css
    Example2.html
```

Using this plugin the result will be
```
Example.Vue
Example2/
    Example2.vue
    Example2.js
    Example2.css
    Example2.html
```

If I use hot or watch with webpack 4 and I modify Example2.js/css/html everything is ok but if I modify Example.vue the watcher don't compile this file.

If I disable the plugin, the watcher works again.

### PR DESCRIPTION
**Pull request by CKGrafico**
Fix #16 

PLEASE: Review first the other PR https://github.com/pksunkara/vue-builder-webpack-plugin/pull/18

PS: You can make my contributor if you want xD

### CODE REVIEW

**Comment by pksunkara** [comment]
Can you rebase this now?

File: index.js [hunk 1]
```diff
@@ -1,12 +1,14 @@
 const recursiveRead = require('recursive-readdir');
 const path = require('path');
 const fs = require('fs');
+const VirtualModulePlugin = require('./virtual-module-webpack-plugin');
```
[the remark this answers is not in the document]
Comment (CKGrafico) [reply index.js]: ouch the npm package xD sorry

File: index.js [hunk 2]
```diff
@@ -150,13 +157,13 @@ const buildVues = (callback) => {
...
-  compiler.plugin('watch-run', (compilation, callback) => buildVues(callback));
+  compiler.plugin('run', (compilation, callback) => buildVues(callback, compiler));
+  compiler.plugin('watch-run', (compilation, callback) => buildVues(callback, compiler));
 
   compiler.plugin('after-compile', (compilation, callback) => {
-    // eslint-disable-next-line no-param-reassign
     compilation.fileDependencies = Array.from(compilation.fileDependencies).filter((file) => {
-      if (file.slice(-4) === '.vue') {
+
+      if (createdFiles.some(x => x === file)) {
```
Comment (pksunkara) [inline index.js]: Wouldn't an `includes` or `indexOf` be faster?

**Comment by CKGrafico** [comment]
No more conflicts I hope xD

File: index.js [hunk 3]
```diff
@@ -159,9 +161,9 @@ VueBuilderPlugin.prototype.apply = (compiler) => {
   compiler.plugin('watch-run', (compilation, callback) => buildVues(callback, compiler));
 
   compiler.plugin('after-compile', (compilation, callback) => {
-    // eslint-disable-next-line no-param-reassign
```
Comment (pksunkara) [inline index.js]: This shouldn't be removed.

File: index.js [hunk 4]
```diff
@@ -159,9 +161,9 @@ VueBuilderPlugin.prototype.apply = (compiler) => {
...
     compilation.fileDependencies = Array.from(compilation.fileDependencies).filter((file) => {
-      if (file.slice(-4) === '.vue') {
+
```
Comment (pksunkara) [inline index.js]: Why is there an empty line?

**Comment by pksunkara** [comment]
Still need to address review comments.

File: index.js (hunk 4 above)
Comment (CKGrafico) [reply index.js]: Not sure if I haveunderstood well, is now ok?

File: index.js (hunk 2 above)
Comment (pksunkara) [reply index.js]: @CKGrafico What about this comment?

Comment (CKGrafico) [reply index.js]: yes https://jsperf.com/includes-vs-some one minute

Comment (CKGrafico) [reply index.js]: check now :)

**Pull request merged**